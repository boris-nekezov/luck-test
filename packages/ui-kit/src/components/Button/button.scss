@import '../../styles/app.scss';

.button {
	@include theme-switcher() {
		border: 1px solid $border-secondary--light;
		background: $button-background--light;
		color: $text-color--light;
		&:hover {
			color: $base-color-white;
			background: $active-primary-color--light;
		}
	}
	padding: 10px 30px;
	border: 1px solid $border-secondary--dark;
	border-radius: 5px;
	background: $base-background--dark;
	color: $text-color--dark;
	cursor: pointer;
	&:hover,
	&.active {
		color: $base-color-white;
		background: $active-secondary-color--dark;
	}
}
