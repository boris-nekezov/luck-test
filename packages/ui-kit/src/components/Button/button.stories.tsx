import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Button } from './Button';

export default {
  title: 'Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const ButtonBasic = Template.bind({});

ButtonBasic.args = {
  children: 'Some button'
};

ButtonBasic.argTypes = {
  handleClick: { action: 'handleClick' },
}