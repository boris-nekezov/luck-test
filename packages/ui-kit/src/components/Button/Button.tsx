import './button.scss';

export interface ButtonProps {
  type?: 'button' | 'submit';
  handleClick?: () => void;
  disabled?: boolean;
  children?: React.ReactNode,
}

export const Button = (props: ButtonProps) => {
  const {
    type = 'button',
    handleClick,
    disabled,
    children,
    ...spreadAttributes
  } = props;

  return (
    <button
      type={type}
      className='button'
      onClick={handleClick}
      disabled={disabled}
      {...spreadAttributes}
    >
      {children}
    </button>
  )
}
