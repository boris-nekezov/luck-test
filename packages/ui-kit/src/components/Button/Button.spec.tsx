import React from 'react';
import { render } from '@testing-library/react';
import { Button } from './Button';

describe('Given `Button` component', () => {
  const mockFn = jest.fn();
  let wrapper: ReturnType<typeof render>;


  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render(<Button
        type="button"
        handleClick={mockFn}
      />);
    });

    it('should match snapshot', () => {
      expect(wrapper.baseElement).toMatchSnapshot();
    })
  })
});
