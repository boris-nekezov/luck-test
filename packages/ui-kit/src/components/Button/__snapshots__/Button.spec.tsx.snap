// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given \`Button\` component when it is rendered should match snapshot 1`] = `
<body>
  <div>
    <button
      class="button"
      type="button"
    />
  </div>
</body>
`;
